sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -546,6 +547,20 @@ abstract class DataWritingCommandMeta[INPUT <: DataWritingCommand](
   override val childDataWriteCmds: Seq[DataWritingCommandMeta[_]] = Seq.empty
 
   override def tagSelfForGpu(): Unit = {}
+
+  // Check whether data type of intput/output contains time zone.
+  // Currently only UTC timezone is supported for [[DataWritingCommand]]
+  def timezoneCheck(): Unit = {
+    val types = (wrapped.inputSet.map(_.dataType) ++ wrapped.outputSet.map(_.dataType)).toSet
+    if (types.exists(GpuOverrides.isOrContainsTimestamp(_))) {
+      if (!GpuOverrides.isUTCTimezone()) {
+        willNotWorkOnGpu("Only UTC timezone is supported for " +
+          "GpuCreateHiveTableAsSelectCommand. " +
```

Comment:
This appears to be called from other commands too. Should that be parameterized?

Reply:
updated.